### DISCUSSION TITLE
Should we drop the Ruby Standalone from this package?

### DISCUSSION THREAD

**Discussion by mefellows** [pact-foundation/pact-node#423] [General]
:wave: hello Pact JS-ers.

We have an issue at the moment (https://github.com/pact-foundation/pact-js-core/issues/416).

The TL;DR is that we package the Ruby standalone into the `pact-core` package, which is a dependency for `pact` to work, and these ruby binaries only work on a subset of OSs now, whereas we support a broader set in Pact JS (notably, arm/aarch64). 

This means that the package can only be run on the lowest common denominator OS/arch combinations, causing issues such as #416 

I’m proposing one of two things:

1. Move the bits of the code out of pact-core into pact, leaving only the bits in pact-core that use the Ruby standalone, and also dropping it as a necessary dependency of Pact JS
2. Remove all traces of the Ruby binary related code in pact-core , and bump a major version.

### Option 1

**Pros**
1. Faster build and release times for `pact` (the thing 99% of our users actually need)
2. Allows teams to install `pact-core` if they still want to get the bundled Ruby CLI

**Cons**
1. Makes the wrapped FFI interface less re-usable (could be solved by creating _yet another_ package)

### Option 2

**Pros**
1. Makes the FFI interface more re-usable
2. Reduces the need to maintain an API we don't want to encourage (the JS wrappers of the Ruby binaries)

**Cons**

1. Removes a nice means to bundle the Ruby CLI into a project (could be mitigated in other ways that benefit the wider ecosystem)

**Comment by TimothyJones-DD** [comment]
I like option 2, but I would also say that we should move the ruby parts into `pact-cli`, and continue to expose the binstubs

**Reply by mefellows** [reply]
Yep, I think that's a reasonable option also. I do think keeping the FFI separate will yield longer term benefits, albeit right now we don't have any immediate use cases.

**Reply by TimothyJones** [reply]
I think it depends how it’s structured. Ideally the only thing in the core is code to do with the FFI- which would mean that the Pact-JS DSL could be used with other pact cores or frameworks, or that replacing it in the future would be easy.

However, if a lot of core concepts end up leaking into the DSL (like plugin implementation detail) there might not be much benefit

**Comment by YOU54F** [comment]
done as part of https://github.com/pact-foundation/pact-js-core/issues/488
